code/0010-define-calculations-of-existential-deposit.md

```diff
@@ -0,0 +1,149 @@
+our native asset'
+
+The ED of any asset should be defined as follows:
+```
+local_ed = native_token_ed * foreign_asset_ratio
+```
+
+This method of setting ED ensures requirement 2.
+
+Example
+```
+token_a_local_ed = 100_000_000
+token_a_asset_ratio = Ratio(1_000_000_000, 1_000_000_000_000)
+token_b_local_ed = 100_000
+token_a_asset_ratio = Ratio(1_000_000, 1_000_000_000_000)
+native_token_ed = 100_000_000_000
+
+fn native_to_token(native_balance, token_asset_ratio) {
+  native_balance * token_asset_ratio
+}
+
+fn token_to_native(token_balance, token_asset_ratio) {
+  token_balance / token_asset_ratio
+}
+
+token_a_ed_as_native = token_to_native(token_a_local_ed, token_a_asset_ratio)
+token_b_ed_as_native = token_to_native(token_b_local_ed, token_b_asset_ratio)
+native_as_token_b = native_to_token(native_token_ed, token_b_asset_ratio)
+token_a_as_token_b = native_to_token(token_a_ed_as_native, token_b_asset_ratio)
+
+assert(token_a_ed_as_native == native_token_ed)
+assert(token_b_ed_as_native == native_token_ed)
+assert(token_a_ed_as_native == token_b_ed_as_native)
+assert(native_as_token_b == token_a_as_token_b)
+```
+
+### Overwriting & Asset Registry Storage
+
+For a token to have its ED be correctly calculated, and therefore to exists on 
+chain, it should have its `foreign_asset_ratio` defined and stored.
+
+If the need arises, an optional ED value can be provided to an asset within
+Asset Registry. This will have priority over the default ED calculation and will
+be evaluated as is.
+
+## Implementation
+
+### Updates to the Functionality of `multi_existential_deposits`
+
+The current implementation of `multi_existential_deposits` does the following:
+  
+  1. If an ED is defined for an asset in Asset Registry, return the ED as is.
+  
+  2. **Otherwise, pull a hard-coded ED from a `match` statement**.
+  
+  3. If no matches are found, return `Balance::MAX` which will prune unknown
+     assets.
+    
+It should be updated as follows:
+
+  1. If an ED is defined for an asset in Asset Registry, return the ED as is.
+  
+  2. **If a asset ratio is defined for an asset in Asset Registry, calculate the
+     standard ED as defined by this document.**
+  
+  3. If no matches are found, return `Balance::MAX` which will prune unknown
+     assets.
+    
+### Existential Deposit Retrieval
+
+For use in the front-end, an RPC will need to be made available that exposes the
+functionality of `multi_existential_deposits`.
+    
+### Asset Registry
+
+**Note**: If AssetRegistry storage is already in use on Picasso, we will need to
+perform a storage migration in the next runtime upgrade for all of these changes
+to be possible.
+
+The function `AssetRegistry::register_asset` should be updated with the 
+following signature:
+
+```rust
+pub fn register_asset(
+  origin: OriginFor<T>,
+  location: T::ForeignAssetId,
+  ed: Option<T::Balance>, // Used to be required
+  ratio: Ratio, // Used to be optional
+  decimals: Option<Exponent>,
```

Comment:
For what we discussed yesterday, we should also make `decimals` required no?

Reply:
decimals no - because `assets` pallet allows assets without decimals.

Reply:
i think we should not prevent us receiving such tokens